examples/creating-cert-chain-with-validation.cpp

```diff
@@ -0,0 +1,247 @@
+/*
+ * #%L
+ * %%
+ * Copyright (C) 2022 BMW Car IT GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+#include <iostream>
+#include <chrono>
+#include <thread>
+
+#include <mococrw/x509.h>
+#include <mococrw/ca.h>
+#include <mococrw/basic_constraints.h>
+#include <mococrw/key_usage.h>
+
+using namespace mococrw;
+
+static const DigestTypes digestType = DigestTypes::SHA512;
+
+/*
+* Create root CA certificate with given keypair
+*/
+X509Certificate createRootCaCertificate(const AsymmetricKeypair &rootKey)
+{
+    // Set the CA to true and path length to 1 (only one intermediate CA)
+    // This will be the actual X509 content of the root CA certificate
+    BasicConstraintsExtension rootCertConstraint{true, 1};
+
+    // "Allow" certificate and certificate revocation list signing
+    const auto rootCaKeyUsage = KeyUsageExtension::Builder{}.keyCertSign().cRLSign().build();
+
+    auto rootCaSelfSignParams = CertificateSigningParameters::Builder{}
+            .certificateValidity(Asn1Time::Seconds(120))
+            .digestType(digestType)
+            .addExtension(rootCertConstraint)
+            .addExtension(rootCaKeyUsage)
+            .build();
+
+    // These are all X509 cert details supported by mococrw
+    auto rootCertDetails = DistinguishedName::Builder{}
+            .commonName("ImATeapot")
+            .countryName("DE")
+            .organizationName("Linux AG")
+            .organizationalUnitName("Linux Support")
+            .pkcs9EmailAddress("support@example.com")
+            .localityName("oben")
+            .stateOrProvinceName("nebenan")
+            .serialNumber("08E36DD501941432358AFE8256BC6EFD")
+            .givenName("Hot Teapot")
+            .userId("1000")
+            .title("Teapots contain tea")
+            .build();
+
+    return CertificateAuthority::createRootCertificate(
+                rootKey,
+                rootCertDetails,
+                // serial number of the root certificate
+                0,
+                rootCaSelfSignParams);
+}
+
+/*
+* Define CA Class for root CA based on root CA Certificate and keypair with random CA settings
+*/
+CertificateAuthority defineRootCa(const X509Certificate &rootCaCert,
```

Comment:
Define seems to not quite represent what is going on here. Maybe rather `configure` or `build`?
Moreover, it would be nice to see an explanation here that MoCoCrW has the `CertificateAuthority` class to provide an interface for CAs to configure how certificates should be issued. + Explanation that they are configured how they issue certificates (as in, which extensions etc should be present in the certificates). They can then be used to sign CSRs. What information is taken from the CSR? What is configured in the CA object?